core/proto/accountmanager/v1/accountmanager.proto

```diff
@@ -0,0 +1,156 @@
+  bool add_signing_keys = 7;
+}
+
+// UpdateUserRequest sets the data field for the account.
+// The account must have the service's current signing key in its list of
+// authorized_keys in order to succeed.
+message UpdateUserRequest {
+  // account contains data which will be applied to the account.
+  keytransparency.v1.proto.User account = 2;
+  // update_mask specifies which fields of account to update.
+  // For example: "data" or "authorized_keys"
+  google.protobuf.FieldMask update_mask = 3;
+}
+
+// BatchCreateUserRequest creates multiple accounts all at once.
+message BatchCreateUserRequest {
+   // domain_id identifies the domain.
+   string domain_id = 1;
+   // app_id identifies the application.
+   string app_id = 2;
+  // accounts is the set of accounts to create.
+  repeated keytransparency.v1.proto.User accounts = 3;
+  // add_signing_keys specifies whether to add this service's signing_keys to the set of authorized_keys.
+  // This must be set to true if any further operations from this API are meant to succeed. 
+  // If set to false, there must be at least one key in authorized_keys.
+  bool add_signing_keys = 4;
+}
+
+// BatchCreateUserRequest creates multiple accounts at once.
+message BatchCreateUserResponse {
+  // accounts returns the list of created accounts. 
+  repeated keytransparency.v1.proto.User accounts = 2;
+} 
+
+// BatchUpdateUserRequest creates multiple accounts all at once.
+message BatchUpdateUserRequest {
+   // domain_id identifies the domain.
+   string domain_id = 1;
+   // app_id identifies the application.
+   string app_id = 2;
+  // accounts is the set of accounts to create.
+  repeated keytransparency.v1.proto.User accounts = 3;
+  // update_mask specifies which fields of account to update.
+  // For example: "data" or "authorized_keys"
+  google.protobuf.FieldMask update_mask = 4;
+}
+
+// BatchUpdateUserRequest creates multiple accounts at once.
+message BatchUpdateUserResponse {
+  // accounts returns the list of created accounts. 
+  repeated keytransparency.v1.proto.User accounts = 2;
+} 
+
+
+// The UserManagerService holds a set of signing keys which can be used to 
+// sign account updates. 
+// - This service must be registered with the server in order to submit account 
+// create or update requests. 
+// - The update requests this service generates will only be valid for accounts
+// that have added this service's signing keys to their list of authorized_keys. 
+// - This service should carefully controll and log access to it's APIs.
+// - This service should rotate its signing keys periodically.
+//   New signing keys should be published far enough in advance that they can be added
+//   by everyone before they are used to avoid service disruption. 
+// 
+// Resources:
+// - domains/apps/keyset
+//   Each combination of domain_id and app_id has it's own keyset.
+// - domains/apps/users
+//   Users can be created and updated through this endpoint.
+service AccounManagerService {
```

Comment:
Just to confirm, will this play well when importing?

Reply:
In what way are you thinking it could go wrong?  
It's certainly my intention for it to work well when importing :-)